File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for(int i : nums){
            set.add(i);
        }
        int maxCount = 0;
        for(int num : set){
            if(!set.contains(num - 1)){
                int current = num;
                int count = 1;
                while(set.contains(current + 1)){
                    count++;
                    current++;
                }
                maxCount = Math.max(maxCount, count);
            }
        }
        return maxCount;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: list[int]) -> int:
        nums = sorted(nums)
        
        Max = 0
        left = 0
        right = 0
        while right < len(nums):
            if right > 0 and nums[right] == nums[right - 1] + 1:
                Max = max(Max, right - left + 1)
                right += 1
            elif right > 0 and nums[right] == nums[right - 1]:
                right += 1
                left += 1
            else:
                left = right
                Max = max(Max, 1)
                right += 1
        return Max

